code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1006,6 +1007,11 @@ func (sc *StorageSmartContract) updateAllocationRequestInternal(
 		}
 
 		for i, bd := range alloc.BlobberDetails {
+			if bd.Terms.WritePrice <= blobbers[i].Terms.WritePrice &&
+				bd.Terms.ReadPrice <= blobbers[i].Terms.ReadPrice {
```

Comment:
Can we treat read price and write price independently. 
* Decrease write pice if blobber terms decreae write price. 
* Decrease read pice if blobber terms decreae read price.

Reply:
Yes that should be better, making changes.